Automattic/wp-calypso
# Site transfers: Refactor site transfer screen

Related to https://github.com/Automattic/dotcom-forge/issues/2553

## Proposed Changes

This pr refactors the site transfer screen to include the design changes mentioned in the above ticket:

- Add a new **Domains** section
- Add a new **Upgrades** section
- Add a new **Content and ownership** section

**Upgrades** section is shown only if the user has available upgrades (anything paid, like a plan etc)
**Domains** section changes it's content based on if the user has custom domains, or not.

## Testing Instructions


### Test 1: Simple site with upgrades and custom domains

1. Create a new simple site, with a business plan.
2. Buy a new domain for this site (eg: a `.blog`
3. Navigate to `Settings > General > Transfer your site`
4. Enter a new user to transfer your site.

Ensure that you see a component like the one below:
![upgrades_domains_transfer](https://github.com/Automattic/wp-calypso/assets/497103/a2c701eb-492d-47be-b986-07d42c46e76b)
5. Check all checkboxes and start transferring your site

Ensure that you see a component like the one below:
![transfer_complete](https://github.com/Automattic/wp-calypso/assets/497103/3c08d68a-d7e8-4d53-a480-6e850b98be9e)


### Test 2: Simple site with no upgrades and no custom domains

1. Create a new simple site, with a personal plan.
3. Navigate to `Settings > General > Transfer your site`
4. Enter a new user to transfer your site.

Ensure that you see a component like the one below:
![no_upgrades_no_domains_transfer](https://github.com/Automattic/wp-calypso/assets/497103/16ec10e0-aeaf-4b81-92dd-54336604b109)

5. Check all checkboxes and start transferring your site

Ensure that you see a component like the one below:

![transfer_complete](https://github.com/Automattic/wp-calypso/assets/497103/3c08d68a-d7e8-4d53-a480-6e850b98be9e)

### Test 3: Error in transfer

1. This change must be tested locally.
2. Create a new simple site, with a personal plan.
3. Navigate to `Settings > General > Transfer your site`
4. Enter a new user to transfer your site.
5. On [this line](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/site-settings/site-owner-transfer/use-start-site-owner-transfer.ts#L30) add:
```
.then(() => { throw new Error('test') } )
```
6. Check all checkboxes and start transferring your site

Ensure that you see a component like the one below:
![transfer_error](https://github.com/Automattic/wp-calypso/assets/497103/12c6a6d1-90cd-43c6-b92f-4bd14d1a1cc1)



**NOTE**
After discussions with @katinthehatsite I changed the `font-weight` to match Figma designs. I use `Chrome` browser on `Linux`, and after those changes, for me, the bold words don't have any visible differences compared to non-bold ones. I don't know if this is an OS issue or not, I just put it as a note here, in case we want to revisit this.



## Pre-merge Checklist
